src/Microsoft.AspNetCore.SignalR.Client.TS/Connection.ts

```diff
@@ -26,12 +28,12 @@ export class Connection {
         this.connectionState = ConnectionState.Disconnected;
     }
 
-    async start(transportName: string = 'webSockets'): Promise<void> {
+    async start(transportName: string = TransportType.webSockets): Promise<void> {
         if (this.connectionState != ConnectionState.Disconnected) {
             throw new Error("Cannot start a connection that is not in the 'Disconnected' state");
         }
 
-        this.transport = this.createTransport(transportName);
+        this.transport = new TransportFactory(this.httpClient).create(transportName);        
```

Comment:
I have some doubts here 
 - we create transports only in one place - why do we need the factory (which is public module)? What's the advantage over the factory function that did just that, was "private" and did not add more to the object state?
 - because the factory is now created in the ctor we need to keep track of it in yet another class variable (referred to as "object state" above)
 - I am also sure if the factory will help in case of custom transports. I would imagine that it would be much easier for the user just to create a custom transport and pass it (when supported) instead of creating a factory that creates a custom transport especially if you think about the dependencies custom transport will require